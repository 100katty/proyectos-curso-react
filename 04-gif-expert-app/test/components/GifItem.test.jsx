import { render , screen ,fireEvent} from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas el component GifItem', () => {
    const title = 'hola'
    const url = 'https://jestjs.io/';
    test('debe de hacer con el snapshot', () => {

        const {container} = render(<GifItem title={title} url={url}  />)
        expect(container).toMatchSnapshot();
       
    });

    test('debe de mostrar la imagen con el url y al alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();

        //expect(screen.getByRole('img').src).toBe(url)

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
    
   
    
});

