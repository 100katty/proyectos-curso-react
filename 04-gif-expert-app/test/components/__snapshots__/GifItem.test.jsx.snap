// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas el component GifItem debe de hacer con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola"
      src="https://jestjs.io/"
    />
    <p>
      hola
       
    </p>
  </div>
</div>
`;
